nsv13/code/controllers/subsystem/overmap_mode.dm

```diff
@@ -58,50 +58,58 @@ SUBSYSTEM_DEF(overmap_mode)
 
 	for(var/D in subtypesof(/datum/overmap_gamemode))
 		var/datum/overmap_gamemode/N = new D()
-		mode_cache += N
+		mode_cache[D] = N
 
 	var/list/mode_pool = mode_cache
 
-	for(var/datum/overmap_gamemode/M in mode_pool)
-		if(M.whitelist_only) //Remove all of our only whitelisted modes
+	for(var/M in mode_pool)
+		var/datum/overmap_gamemode/GM = mode_pool[M]
+		if(GM.whitelist_only) //Remove all of our only whitelisted modes
+			QDEL_NULL(mode_pool[M])
 			mode_pool -= M
 
 	if(SSmapping.config.omode_blacklist.len > 0)
 		if(locate("all") in SSmapping.config.omode_blacklist)
 			mode_pool = list() //Clear the list
 		else
 			for(var/S in SSmapping.config.omode_blacklist) //Grab the string to be the path - is there a proc for this?
-				var/datum/overmap_gamemode/B = text2path("/datum/overmap_gamemode/[S]")
+				var/B = text2path("/datum/overmap_gamemode/[S]")
+				QDEL_NULL(mode_pool[B])
 				mode_pool -= B
 
 	if(SSmapping.config.omode_whitelist.len > 0)
 		for(var/S in SSmapping.config.omode_whitelist) //Grab the string to be the path - is there a proc for this?
-			var/datum/overmap_gamemode/W = text2path("/datum/overmap_gamemode/[S]")
-			mode_pool += W
+			var/W = text2path("/datum/overmap_gamemode/[S]")
+			mode_pool[W] = new W()
 
 	for(var/mob/dead/new_player/P in GLOB.player_list) //Count the number of connected players
 		if(P.client)
 			player_check ++
 
-	for(var/datum/overmap_gamemode/M in mode_pool) //Check and remove any modes that we have insufficient players for the mode
-		if(player_check < M.required_players)
+	for(var/M in mode_pool) //Check and remove any modes that we have insufficient players for the mode
+		var/datum/overmap_gamemode/GM = mode_pool[M]
+		if(player_check < GM.required_players)
+			QDEL_NULL(mode_pool[M])
 			mode_pool -= M
 
-	if(mode_pool.len <= 0) //If the pool is empty, we set the default
-		mode = /datum/overmap_gamemode/patrol //Holding that as the default for now - REPLACE ME LATER
-		message_admins("Error: mode section pool empty - defaulting to PATROL")
-		log_game("Error: mode section pool empty - defaulting to PATROL")
-
-	else //Here we need to generate a ticket system that pulls from the config at a future date
-
+	var/mode_type = /datum/overmap_gamemode/patrol
+	if(mode_pool.len)
```

Comment:
I only flipped these because I thought I would use mode_type at the end but then I didn't vOv